crates/subspace-node/src/chain_spec.rs

```diff
@@ -356,6 +359,9 @@ pub fn dev_config() -> Result<GenericChainSpec, String> {
                     permissioned_action_allowed_by: PermissionedActionAllowedBy::Accounts(vec![
                         sudo_account.clone(),
                     ]),
+                    #[cfg(feature = "runtime-benchmarks")]
+                    genesis_domains: vec![],
+                    #[cfg(not(feature = "runtime-benchmarks"))]
```

Comment:
Looks like other `runtime-benchmarks` conditional compile is due to this, and I guess this is because the memory allocation error during benchmark.

For which I have researched on the upstream codebase, although I haven't located the root cause, but the error can be bypassed by benchmarking with `--genesis-builder=none`, which means not running the benchmark with any pre-exist genesis state, and it is okay since in the benchmark we don't expect any pre-exist genesis state.

Reply:
For now I would leave it as is.  I have not tested with no genesis-builder but something we can update later though I see no reason why genesis state must be skipped here.

Reply:
if you have a strong opinion on this. Then that change has to wait for a bit with me focussing on more priority XDM issue. 
I may mostly handle this in different PR once I get some time to test and in the meantime if you have found a root cause, let me know